Refactor EXTENSION_AUTOLOADING Make and CMake settings

carlopi opened the pull request:
Previously, there were two independent settings ENABLE_EXTENSION_AUTOLOADING and ENABLE_EXTENSION_AUTOINSTALL, defaulting to 1 Now there is only one, EXTENSION_AUTOLOADING, that takes either 'none', 'load', or 'full'.

Currently default behaviour falls back to 'none', but emits a CMake warning. I think it would be positive to move all our builds to specify explicitly the setting for autoloading.

At C++ level logic remains unchanged, and defaults also remains unchanged.

carlopi commented:
Failures are due to vcpkg not finding a package (Linux Extensions + Python Linux Extensions) plus unrelated CodeCoverage

Mytherin commented:
Can you switch the configuration for this to be on by default instead (maybe unless `DUCKDB_DISABLE_EXTENSION_LOAD` is enabled?).

carlopi committed:
Refactor EXTENSION_AUTOLOADING Make and CMake settings

Previously, there were two independent settings ENABLE_EXTENSION_AUTOLOADING and ENABLE_EXTENSION_AUTOINSTALL, defaulting to 1
Now there is only one, EXTENSION_AUTOLOADING, that takes either 'none', 'load', or 'full'.

Currently default behaviour falls back to 'none', but emits a CMake warning. I think it would be
positive to move all our builds to specify explicitly the setting for autoloading.

At C++ level logic remains unchanged, and defaults also remains unchanged.

Mytherin commented:
Closing
